### PR TITLE
Extend upgrading guide

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -36,6 +37,8 @@ Multiple backwards incompatible changes
 ### Core packages
 
 - `_HarHatHelperIntegration` has been renamed to `HardHatHelperIntegration` in `packages/valory/skills/abstract_round_abci/test_tools/integration.py`
+- `ResetPauseABCIApp` has been renamed to `ResetPauseAbciApp` in `packages/valory/skills/reset_pause_abci/rounds.py`
+- `AppSpecs` not use dataclasses in order to encapsulate data, which means `response_key`, `response_info`, `error_data`, `retries_attempted` and `backoff_factor` are no longer defined as part of the `AppSpecs` implementation and has been replaced with `response_info` and `retries_info` class variables.
```
Comment (Adamantios) [inline docs/upgrading.md]:
```suggestion
- `ApiSpecs` now use dataclasses in order to encapsulate data, which means that `response_key`, and `response_type` can be accessed via the `response_info` instance attribute. Moreover, this attribute now includes the new `response_index` and `error_key`, `error_index`, `error_type`, `error_data`. The `retries_info` instance attribute gives access to the `retries_attempted` and `retries` which are now public and also includes the new `backoff_factor`.
```

File: docs/upgrading.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ Multiple backwards incompatible changes
 - `autonomy analyse abci check-app-specs` and `autonomy analyse generate-app-specs` has been merged into `autonomy analyse fsm-specs`
   - The usage of `--infile` flag has been deprecated and replaced with the usage of the `--package` flag
   - Input format for `--app-class` has been changed from `packages.author.skills.skill_name.rounds.SomeAbciApp` to `SomeAbciApp`
+  - Users will have to manually switch to the newer version
```
Comment (Adamantios) [inline docs/upgrading.md:19]:
```suggestion
  - Users will have to manually switch to the newer version, i.e., remove the path prefix in all their fsm specs. For example an FSM spec with a `label: packages.author.skills.my_abci.rounds.MyAbciApp` now becomes `label: MyAbciApp`

**Review by Adamantios — APPROVED** [review]

File: docs/upgrading.md [hunk 3]
```diff
@@ -36,6 +37,8 @@ Multiple backwards incompatible changes
 ### Core packages
 
 - `_HarHatHelperIntegration` has been renamed to `HardHatHelperIntegration` in `packages/valory/skills/abstract_round_abci/test_tools/integration.py`
+- `ResetPauseABCIApp` has been renamed to `ResetPauseAbciApp` in `packages/valory/skills/reset_pause_abci/rounds.py`
+- `ApiSpecs` now use `dataclasses` in order to encapsulate data, which means that `response_key`, and `response_type` can be accessed via the `response_info` instance attribute. Moreover, this attribute now includes the new `response_index` and `error_key`, `error_index`, `error_type`, `error_data`. The `retries_info` instance attribute gives access to the `retries_attempted` and `retries` which are now public and also includes the new `backoff_factor`.
```
Comment (Adamantios) [inline docs/upgrading.md:42]:
```suggestion

### Test tools

- The `packages.valory.skills.abstract_round_abci.test_tools.apis` module which contained the `DummyMessage` has been removed. `MagicMock` can be used instead. For example, `DummyMessage(my_test_body)` can be converted to `MagicMock(body=my_test_body)`.
```

**Review by Adamantios — CHANGES_REQUESTED** [review]
Found one more.

File: docs/upgrading.md [hunk 4]
```diff
@@ -36,7 +37,12 @@ Multiple backwards incompatible changes
...
 
+### Test tools
+
+- The `packages.valory.skills.abstract_round_abci.test_tools.apis` module which contained the `DummyMessage` has been removed. `MagicMock` can be used instead. For example, `DummyMessage(my_test_body)` can be converted to `MagicMock(body=my_test_body)`.
 ## `v0.3.4` to `v0.3.5`
```
Comment (DavidMinarsch) [inline docs/upgrading.md:46]:
```suggestion
- The `packages.valory.skills.abstract_round_abci.test_tools.apis` module which contained the `DummyMessage` has been removed. `MagicMock` can be used instead. For example, `DummyMessage(my_test_body)` can be converted to `MagicMock(body=my_test_body)`.

## `v0.3.4` to `v0.3.5`
```

**Pull request merged**